### ISSUE TITLE
null pointer exception related to hidden columns

### ISSUE THREAD

**Issue by robrez** [vaadin/vaadin-grid#1765]
### Description

Am encountering an exception here:

```javascript
  __setColumnTemplateOrRenderer(template, renderer, cells) {
    if (template && renderer) {
      throw new Error('You should only use either a renderer or a template');
    }

    cells.forEach(cell => {
      try {
      const model = this._grid.__getRowModel(cell.parentElement);         // <<<<<<<
```

```bash
Uncaught TypeError: Cannot read property 'index' of null
    at HTMLElement.__getRowModel (vaadin-grid.js:814)
    at vaadin-grid-column.js:257
    at Array.forEach (<anonymous>)
    at HTMLElement.__setColumnTemplateOrRenderer (vaadin-grid-column.js:256)
    at HTMLElement._setBodyTemplateOrRenderer (vaadin-grid-column.js:282)
    at Object.runMethodEffect [as fn] (property-effects.js:1038)
    at runEffects (property-effects.js:140)
    at HTMLElement._propertiesChanged (property-effects.js:1922)
    at HTMLElement._flushProperties (properties-changed.js:384)
    at HTMLElement._invalidateProperties (property-effects.js:1748)
```

### Expected outcome
No exception

### Actual outcome
Exception

### Live Demo

TODO, I have been struggling to make an isolated repro, I will try to update this after work hours. I can provide detail below regarding the general setup... maybe someone on the v-team will know some obvious detail

### Steps to reproduce

Using a grid with:

- height-by-rows
- column + header renderers for every column
- direct array data providing via items property
- hiding columns dynamically via hidden attr on some columns

Have tested in a few 5.6.x and 5.7.0-x releases and am getting the exception in all of them.

### Browsers Affected

- [x] Chrome

**Comment by knjk04** [comment]
I've also had this issue with Firefox

**Comment by epiera** [comment]
+1

**Comment by enver-haase** [comment]
@robrez I have an isolated issue here, kindly donated by a customer. See VS-3727 if you can.

**Comment by enver-haase** [comment]
Dear subscribers, do your grids involve re-ordering columns programmatically ?

**Comment by robrez** [comment]
Just looking at the commits in `v5.7.10` (not yet published to npm).  Was thinking there was a good chance this issue would be resolved via https://github.com/vaadin/vaadin-grid/commit/e52c066c14a73e71ef1738493210478122b1d661

I have re-tested my (private) repo with that patch applied and have confirmed it prevents the exception I reported originally and things seem to render as expected

**Comment by sosa-vaadin** [comment]
I'm closing this issue since it seems to be fixed by `v5.7.11` which has been released.
